src/coreclr/tools/Common/TypeSystem/Common/MetadataVirtualMethodAlgorithm.cs

```diff
@@ -964,7 +944,11 @@ public static MethodDesc ResolveVariantInterfaceMethodToStaticVirtualMethodOnTyp
                     {
                         // Attempt to resolve on variance matched interface
                         MethodDesc runtimeInterfaceMethod = runtimeInterfaceType.FindMethodOnExactTypeWithMatchingTypicalMethod(interfaceMethod);
-                        resolvedMethodOnType = TryResolveVirtualStaticMethodOnThisType(typeToCheck, runtimeInterfaceMethod);
+
+                        if (runtimeInterfaceMethod != null)
```

Comment:
How can this end up being null? On line 936 above we validated this is the same type (just a different generic instantiation of it).

Reply:
Based on your feedback I have reverted all changes to the <code>MetadataVirtualMethodAlgorithm.cs</code> file, apparently the relevant logic has already been added for NativeAOT - after all, the principles of virtual method resolution shouldn't differ between NativeAOT and Crossgen2.